### PR TITLE
i198: Update UI in AuthPage and EmailAuthOptionsPage

### LINKED ISSUE #198: Update AuthPage and EmailAuthOptionsPage to match new designs
#### AuthPage 
- [x] email auth options button -> FAB 
- [x] other auth options button 
   - [x] change to a FAB (with non-default provider icon) 
   - [x] no navigation, just initiate sign in 
- [x] add twitter FAB (disabled for now) 
- [x] add facebook FAB (disabled for now) 

#### OtherAuthOptionsPage
- [x] remove OtherAuthOptionsPage 

#### EmailAuthOptionsPage
- [x] remove chips 
- [x] add text and links to move between options

### PR DESCRIPTION
**Pull request by nickmeinhold**
fixes #198

### CODE REVIEW

File: lib/extensions/models_extensions.dart [hunk 1]
```diff
@@ -26,8 +24,6 @@ extension NavigatorEntriesExt on BuiltList<PageData> {
         key: ValueKey(AuthOrMain), child: AuthOrMain()),
     EmailAuthPageData(): MaterialPage<EmailAuthOptionsPage>(
         key: ValueKey(EmailAuthOptionsPage), child: EmailAuthOptionsPage()),
-    OtherAuthOptionsPageData(): MaterialPage<OtherAuthOptionsPage>(
```
Comment (nickmeinhold) [inline lib/extensions/models_extensions.dart:29]: removed as we don't have a separate page any more

File: lib/models/settings/settings.dart [hunk 2]
```diff
@@ -23,9 +23,9 @@ abstract class Settings implements Built<Settings, SettingsBuilder> {
   static SettingsBuilder initBuilder() {
     return SettingsBuilder()
       ..darkTheme.brightness = ThemeBrightness.dark
-      ..darkTheme.colors = ThemeColors.crowdleague.toBuilder()
```
Comment (nickmeinhold) [inline lib/models/settings/settings.dart:26]: changed so the app is in greyscale mode to start with, should have always been this way

File: lib/widgets/auth/auth_page.dart [hunk 3]
```diff
@@ -65,12 +64,15 @@ class PageContents extends StatelessWidget {
...
                 PlatformSignInButton(),
-                SizedBox(
-                  height: 10,
-                ),
+                SizedBox(height: 30),
                 Row(
                   mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-                  children: [EmailOptionsButton(), OtherOptionsButton()],
+                  children: [
+                    EmailOptionsButton(),
```
Comment (nickmeinhold) [inline lib/widgets/auth/auth_page.dart:71]: added FABs for sign in options - Twitter and Facebook are disabled and greyed out for now

File: lib/widgets/auth/auth_page.dart [hunk 4]
```diff
@@ -159,69 +161,78 @@ class PlatformSignInButton extends StatelessWidget {
   }
 }
 
-class OtherOptionsButton extends StatelessWidget {
+class OtherDefaultProviderButton extends StatelessWidget {
```
Comment (nickmeinhold) [inline lib/widgets/auth/auth_page.dart:164]: Just replaced a normal button (with icon and text) that navigated to a page with a single sign in button to a FAB that does the sign in directly

File: lib/widgets/auth/auth_page.dart [hunk 5]
```diff
@@ -159,69 +161,78 @@ class PlatformSignInButton extends StatelessWidget {
...
+        elevation: 1,
+        mini: true,
+        onPressed: () => context.dispatch(SignInWithApple()));
   }
 }
 
 class EmailOptionsButton extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return ButtonTheme(
```
Comment (nickmeinhold) [inline lib/widgets/auth/auth_page.dart:199]: just a normal button to a FAB, same behaviour

File: lib/widgets/auth/email_auth_options_page.dart [hunk 6]
```diff
@@ -45,14 +43,6 @@ class EmailAuthOptionsPage extends StatelessWidget {
                       child: Column(
                         crossAxisAlignment: CrossAxisAlignment.center,
                         children: [
-                          SizedBox(height: 20),
```
Comment (nickmeinhold) [inline lib/widgets/auth/email_auth_options_page.dart:48]: removal of chip shenanigans

File: lib/widgets/auth/email_auth_options_page.dart [hunk 7]
```diff
@@ -233,52 +224,6 @@ class PasswordSuffixIconButton extends StatelessWidget {
   }
 }
 
-class EmailSignInChip extends StatelessWidget {
```
Comment (nickmeinhold) [inline lib/widgets/auth/email_auth_options_page.dart:236]: more chip removal

File: lib/widgets/auth/email_auth_options_page.dart [hunk 8]
```diff
@@ -364,3 +309,69 @@ class CreateAccountButton extends StatelessWidget {
     );
   }
 }
+
+class SwitchModeText extends StatelessWidget {
```
Comment (nickmeinhold) [inline lib/widgets/auth/email_auth_options_page.dart:313]: adding a text+link combo that does the same as the chips but looks less shit

File: test/widget/auth/auth_page_widget_test.dart [hunk 9]
```diff
@@ -239,5 +239,107 @@ void main() {
       // check correct action is dispatched
       expect(testMiddleware.received(SignInWithGoogle()), true);
     });
+
```
Comment (nickmeinhold) [inline test/widget/auth/auth_page_widget_test.dart:242]: these tests were moved from the deleted "other_auth_options_page_widget_test.dart" as the UI from OtherAuthOptionsPage was moved into AuthPage - I'm pretty sure only some small renaming changes were made and the logic is all exactly the same.

File: test/widget/auth/email_auth_options_page_widget_test.dart [hunk 10]
```diff
@@ -96,7 +96,7 @@ void main() {
     });
 
     testWidgets(
```
Comment (nickmeinhold) [inline test/widget/auth/email_auth_options_page_widget_test.dart:98]: all just renaming changes, no logic

File: test/widget/auth/email_auth_options_page_widget_test.dart [hunk 11]
```diff
@@ -132,10 +132,11 @@ void main() {
...
-        'dispatches UpdateEmailAuthOptionsPage action when tap sign in chip',
+        'dispatches UpdateEmailAuthOptionsPage action on tapping SIGN IN link',
         (WidgetTester tester) async {
       // Setup the app state with expected values
-      final initialAppState = AppState.init();
+      final initialAppState = AppState.init()
+          .rebuild((b) => b..emailAuthOptionsPage.mode = EmailAuthMode.signUp);
```
Comment (nickmeinhold) [inline test/widget/auth/email_auth_options_page_widget_test.dart:139]: actually this was a logic change - it makes sense to me why it passes now but I don't get why it was passing before... I'm happy to leave it as a mystery if you are

File: lib/extensions/models_extensions.dart (hunk 1 above)
Comment (tamari-gray) [reply lib/extensions/models_extensions.dart:29]: noice

File: lib/models/settings/settings.dart (hunk 2 above)
Comment (tamari-gray) [reply lib/models/settings/settings.dart:26]: nice pickup, make those users focus on what matters!

File: lib/widgets/auth/auth_page.dart (hunk 3 above)
Comment (tamari-gray) [reply lib/widgets/auth/auth_page.dart:71]: sweet, theyll be bouncin soon 🏀

File: lib/widgets/auth/auth_page.dart (hunk 4 above)
Comment (tamari-gray) [reply lib/widgets/auth/auth_page.dart:164]: nice!!!

File: lib/widgets/auth/auth_page.dart (hunk 5 above)
Comment (tamari-gray) [reply lib/widgets/auth/auth_page.dart:199]: sweet

File: lib/widgets/auth/email_auth_options_page.dart (hunk 6 above)
Comment (tamari-gray) [reply lib/widgets/auth/email_auth_options_page.dart:46]: goodbye chips 👋

File: lib/widgets/auth/email_auth_options_page.dart (hunk 7 above)
Comment (tamari-gray) [reply lib/widgets/auth/email_auth_options_page.dart:222]: "chip removal" cracks me up

File: lib/widgets/auth/email_auth_options_page.dart [hunk 12]
```diff
@@ -69,10 +61,11 @@ class EmailAuthOptionsPage extends StatelessWidget {
                               autovalidateMode:
                                   vm.autovalidate.toAutovalidateMode(),
                             ),
-                          SizedBox(height: 50),
+                          SizedBox(height: 30),
```
Comment (tamari-gray) [inline lib/widgets/auth/email_auth_options_page.dart:64]: 👀 u touching my layout?

Comment (tamari-gray) [reply lib/widgets/auth/email_auth_options_page.dart:64]: (i dont actually care lol)

File: lib/widgets/auth/email_auth_options_page.dart (hunk 8 above)
Comment (tamari-gray) [reply lib/widgets/auth/email_auth_options_page.dart:301]: chip revolution

File: pubspec.yaml [hunk 13]
```diff
@@ -47,6 +47,11 @@ flutter:
 
   assets:
     - assets/images/logo-300-greyscale.png
+    - assets/images/facebook_logo.png
```
Comment (tamari-gray) [inline pubspec.yaml:50]: great job finding these 🙏

File: test/widget/auth/auth_page_widget_test.dart (hunk 9 above)
Comment (tamari-gray) [reply test/widget/auth/auth_page_widget_test.dart:242]: sweet!

File: test/widget/auth/email_auth_options_page_widget_test.dart (hunk 10 above)
Comment (tamari-gray) [reply test/widget/auth/email_auth_options_page_widget_test.dart:98]: cool 😄

File: test/widget/auth/email_auth_options_page_widget_test.dart (hunk 11 above)
Comment (tamari-gray) [reply test/widget/auth/email_auth_options_page_widget_test.dart:139]: ooh lets discuss @nickmeinhold !

**Review by tamari-gray — APPROVED** [review]
beautiful pr, great job getting those fabs in and cleaning up the auth page / email auth page.
they look really good now! and there should be a github badge for UX points cause we definitely gained some from this!

wanted to have a little discussion about that sign in chip test mysteriously working with logic change, but that doesnt block this pr so im pushing her through!

great job again!

**Pull request merged**